roles/openshift_health_checker/openshift_checks/docker_image_availability.py

```diff
@@ -0,0 +1,148 @@
+        if not registries:
+            return {"failed": True, "msg": "Unable to read docker registry data from host."}
+
+        # inspect all rpm or containerized docker images for every docker registry found
+        while registries and required_images:
+            registry = registries.pop(0)
+            required_images = self.inspect_images(registry, required_images, task_vars)
+
+        # if there are required images left, fail the check
+        if required_images:
+            msg = "Unable to find required images in any docker registries: {imgs}".format(imgs=list(required_images))
+            return {"failed": True, "changed": True, "msg": msg}
+
+        return {"changed": True}
+
+    def get_required_images(self, is_containerized, reg_prefix, openshift_release):
+        if is_containerized:
+            return set(self.containerized_docker_images(reg_prefix, openshift_release))
+
+        return set(self.rpm_docker_images(reg_prefix, openshift_release))
+
+    def containerized_docker_images(self, reg_prefix, version):
+        return [
+            "{image}:{version}".format(image=self.image_from_reg_prefix(reg_prefix), version=version)
+        ]
+
+    @staticmethod
+    def rpm_docker_images(reg_prefix, version):
+        return [
+            "{image_repo}/registry-console:{version}".format(image_repo=reg_prefix["repo"], version=version)
+        ]
+
+    @staticmethod
+    def qualified_docker_images(image_name, version):
+        return [
+            "{image_name}-{component}:{version}".format(**locals())
+            for component in "haproxy-router docker-registry deployer pod".split()
+        ]
+
+    @staticmethod
+    def image_from_reg_prefix(prefix):
+        return "".join([prefix["repo"], "/", prefix["image"]])
+
+    def inspect_images(self, registry, tagged_images, task_vars):
+        failed = set()
+        for image in tagged_images:
+            # the run command pulls the {skopeo_image} if it has not already been pulled.
+            # if pulling this image fails, the current image will be added to the failed set.
+            cmd_str = "docker run {skopeo_image} skopeo inspect docker://{registry}/{tagged_image}".format(
+                skopeo_image=self.skopeo_image,
+                registry=registry,
+                tagged_image=image
+            )
+
+            # use a "docker run ..." command on the remote host to inspect each specific
+            # image inside of a Skopeo image container. If the command's return code is
+            # non-zero, it is safe to assume that an image with the specified tag does not
+            # exist in the given registry.
+            if self.exec_cmd(cmd_str, task_vars)["rc"] != 0:
+                failed.add(image)
+
+        return failed
+
+    def inspect_local_images(self, tagged_images, task_vars):
+        failed = set()
+        for image in tagged_images:
+            cmd = self.module_executor("docker_image_facts", {"name": image}, task_vars)
```

Comment:
> Perhaps you already knew all that and thus was proposing to get facts one by one?

Yes, since I can know that an image was not "found" if the dictionary that is returned contains an empty `images: []` list, it is a safer to execute the module with one image at a time and compile the list available/unavailable images this way, rather than depending on uncertain values from the response's image objects.